cpp/ql/src/semmle/code/cpp/valuenumbering/HashCons.qll

```diff
@@ -738,12 +750,8 @@ private predicate mk_FieldCons(
     e = cal.getFieldExpr(f).getFullyConverted() and
     f.getInitializationOrder() = i and
     (
-      exists(HashCons head, Field f2, HC_Fields tail |
-        hc = hashCons(e) and
-        hcf = HC_FieldCons(c, i - 1, f2, head, tail) and
-        f2.getInitializationOrder() = i - 1 and
-        mk_FieldCons(c, i - 1, f2, head, tail, cal)
```

Comment:
I assume it's deliberate that the call to `getInitializationOrder` has been deleted. It seems to just state a fact that follows from the recursive call to `mk_FieldCons`.

Reply:
Yes, it should be implied by the recursive call. [I can actually simplify a bit further, but was observing odd behaviour, probably due to the off-by-one @rdmarsh2 spotted. I'll test it out and perhaps add another commit.